src/hooks/usePelcro/pelcroActions.js

```diff
@@ -40,6 +40,18 @@ export class PelcroActions {
       return this.set({ view: "login" });
     }
 
+    if (view === "newsletter-update") {
+      const newsletters = window.Pelcro?.uiSettings?.newsletters;
+      const siteHasNewslettersDefined =
+        Array.isArray(newsletters) && newsletters.length > 0;
+      if (
+        !this.get().isAuthenticated() ||
+        !siteHasNewslettersDefined
+      ) {
+        return;
+      }
```

Comment:
We're trying to keep all our actions tested with good coverage. Can we add this test case? If it's not covered?

Reply:
yup it's not covered, will add tests